src/main/java/edu/cornell/kfs/vnd/document/CuVendorMaintainableImpl.java

```diff
@@ -221,5 +235,97 @@ private void setupMultiselectField(MaintenanceDocument document, MaintenanceDocu
         }
         
     }
+    
+	@Override
+	public void doRouteStatusChange(DocumentHeader header) {
+		super.doRouteStatusChange(header);
+		VendorDetail vendorDetail = (VendorDetail) getBusinessObject();
+		WorkflowDocument workflowDoc = header.getWorkflowDocument();
+		boolean existingPaymentWorksVendor = getPaymentWorksVendorService().isExistingPaymentWorksVendorByDocumentNumber(this.getDocumentNumber());
+
+		if (workflowDoc.isFinal()) {
```

Comment:
Sorry for not responding to this question earlier.  I believe we want to it to be for final documents.  After it completes work flow, we need to update corresponding records in paymentworks vendor BO, and send updates to payment works SAS.
This is what UConn was using, and it seems to be working as expected so far.  As we start turning on more functionality and getting purchasing using it, we may need to adjust this logic.